src/Tests/Microsoft.NET.Publish.Tests/RuntimeIdentifiersTests.cs

```diff
@@ -193,6 +193,76 @@ public void PublishWithRuntimeIdentifier(bool publishNoBuild)
             }
         }
 
+        [WindowsOnlyTheory]
+        [InlineData("net7.0", "win-x64", "win-x86", false, false)]
+        [InlineData("net7.0", "win-x64", "win-x86", true, false)]
+        [InlineData("net7.0", "win-x64", "win-x86", true, true)]
+        public void PublishRuntimeIdentifierSetsRuntimeIdentifierAndDoesOrDoesntOverrideRID(string tfm, string publishRuntimeIdentifier, string runtimeIdentifier, bool runtimeIdentifierIsGlobal, bool publishRuntimeIdentifierIsGlobal)
+        {
+            var testProject = new TestProject()
+            {
+                IsExe = true,
+                TargetFrameworks = tfm
+            };
+            testProject.AdditionalProperties["RuntimeIdentifier"] = runtimeIdentifier;
+            testProject.AdditionalProperties["PublishRuntimeIdentifier"] = publishRuntimeIdentifier;
+            testProject.RecordProperties("RuntimeIdentifier");
+
+            List<string> args = new List<string>
+            {
+                $"/p:_IsPublishing=true", // Normally this would be set by the CLI (OR VS Soon TM), but this calls directly into t:/Publish.
+                runtimeIdentifierIsGlobal ? $"/p:RuntimeIdentifier={runtimeIdentifier}" : "",
+                publishRuntimeIdentifierIsGlobal ? $"/p:PublishRuntimeIdentifier={publishRuntimeIdentifier}" : ""
+            };
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject, identifier: $"{publishRuntimeIdentifierIsGlobal}-{runtimeIdentifierIsGlobal}");
+            var publishCommand = new PublishCommand(testAsset);
+            publishCommand
+                .Execute(args.ToArray())
+                .Should()
+                .Pass();
+
+            string expectedRid = runtimeIdentifierIsGlobal ? runtimeIdentifier : publishRuntimeIdentifier;
+            var properties = testProject.GetPropertyValues(testAsset.TestRoot, targetFramework: tfm, runtimeIdentifier: expectedRid);
+            var finalRid = properties["RuntimeIdentifier"];
+
+            Assert.True(finalRid == expectedRid); // This assert is theoretically worthless as the above code will fail if the RID path is wrong.
+        }
+
+        [WindowsOnlyTheory]
+        [InlineData("net7.0", "tizen")] // tizen is an arbitrary nonwindows rid, picked because it will be different from a windows rid.
+        public void PublishRuntimeIdentifierDoesNotOverrideUseCurrentRuntime(string tfm, string publishRid)
```

Comment:
It feels like this behavior is backwards from what it should be.  If you specify both `UseCurrentRuntimeIdentifier` and `PublishRuntimeIdentifier` in the project file, it is probably because you want to build for the development machine, but publish for a different target.

@richlander @baronfel Thoughts?

Reply:
I agree, not sure what I was thinking before. I'll change it for now but leave this thread open.